core/src/main/java/org/mule/config/pool/DefaultThreadPoolFactory.java

```diff
@@ -28,19 +30,20 @@
 
     public ThreadPoolExecutor createPool(String name, ThreadingProfile tp)
     {
+        ThreadingProfile threadingProfileCopy = tp instanceof DirectThreadingProfile ? tp : new ImmutableThreadingProfile(tp);
```

Comment:
This is required because the existing code is so bad, i hate it too.  I suggest we create an issue to improve this code, as there is not time to do this in scope of this issue.